server-spi-private/src/main/java/org/keycloak/utils/CredentialHelper.java

```diff
@@ -105,6 +109,29 @@ public static boolean createOTPCredential(KeycloakSession session, RealmModel re
         return user.credentialManager().isValid(credential);
     }
 
+    /**
+     * Create RecoveryKeys credential either in userStorage or local storage (Keycloak DB)
+     *
+     * @return true if credential was successfully created either in the user storage or Keycloak DB. False if error happened (EG. during HOTP validation)
```

Comment:
The javadoc is wrong because of the copy/paste, the return `true` now makes no sense. Please just create the proper javadoc for the method.